changeset.go

```diff
@@ -72,10 +72,19 @@ func (c Changeset) Constraints() Constraints {
 // Apply mutation.
 func (c *Changeset) Apply(doc *rel.Document, mut *rel.Mutation) {
 	var (
-		pField = doc.PrimaryField()
-		now    = time.Now().Truncate(time.Second)
+		pField        = doc.PrimaryField()
+		mutablePField = false
+		now           = time.Now().Truncate(time.Second)
 	)
 
+	switch c.values[pField].(type) {
+	case int:
+	default:
+		if c.changes[pField] != nil {
+			mutablePField = true
```

Comment:
this actually makes `field != pField` to never be evaluated

Reply:
Yes, I am aware. It looks that this check is only needed to **not** update PK. If PK is not serial and is explicitly changed, then the check is redundant.

Reply:
Or it seems to me so. I have not explored the whole repo yet, and may forget some cases. This change is made to allow explicit change of PK. Accidental mutation would be possible only if is called by user explicitly.

Reply:
I'm sorry, only now I understood the problem. The `field != pField` will not be executed only if pField is non-integer value and is set in the changeset.
Line 81 actually will break out of switch in case of `pField` being an int. There is no fallthrough.